tls/s2n_server_hello.c

```diff
@@ -286,9 +286,6 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
         POSIX_GUARD(s2n_prf_key_expansion(conn));
     }
 
-    /* Choose a default signature scheme */
-    POSIX_GUARD(s2n_choose_default_sig_scheme(conn, &conn->handshake_params.server_cert_sig_scheme, S2N_SERVER));
```

Comment:
Why can we remove choose_default_sig_scheme in this case? Shouldn't we be swapping it out with s2n_signature_algorithms_recv?

Reply:
No, because there's nothing to receive here. Let me see if I can clarify:

For TLS1.3, it's easy. Both the server and client always send their sig_alg choice in the certificate_verify message, and their peer needs the sig_alg to process the certificate_verify message (to verify the signature). So let's just ignore TLS1.3.

For TLS1.2:
* The server sends its sig_alg choice in the key_exchange message, and the client needs the sig_alg to process the key_exchange message. 
* The client sends its sig_alg choice in the certificate_verify message, and the server needs the sig_alg to process the key_exchange message. 

For <TLS1.2:
* The server needs a sig_alg to process the key_exchange message, but just uses the default.
* The client needs a sig_alg to process the certificate_verify message, but just uses the default.

There's no reason for setting the sig_alg on the client when it receives a ServerHello. If it's TLS1.2, it'll receive the real sig_alg with the server key_exchange message later. If it's <TLS1.2, it won't need to use the sig_alg until the server key_exchange message anyway. It just makes the behavior unnecessarily different for TLS1.2 vs <TLS1.2 and client vs server.